pkg/subscriber/helmrepo/helm_subscriber_item.go

```diff
@@ -241,10 +249,89 @@ func (hrsi *SubscriberItem) getHelmRepoIndex(client rest.HTTPClient, repoURL str
 		return nil, "", err
 	}
 
-	err = utils.FilterCharts(hrsi.Subscription, indexfile)
+	err = utils.FilterCharts(sub, indexfile)
 
 	return indexfile, hash, err
 }
+func GetSubscriptionChartsOnHub(hubClt client.Client, sub *appv1.Subscription) ([]*releasev1.HelmRelease, error) {
+	chn := &chnv1.Channel{}
+	chnkey := utils.NamespacedNameFormat(sub.Spec.Channel)
+
+	if err := hubClt.Get(context.TODO(), chnkey, chn); err != nil {
+		return nil, gerr.Wrapf(err, "failed to get channel of subscription %v", sub)
+	}
+
+	repoURL := chn.Spec.Pathname
+	klog.V(2).Infof("getting resource list of HelmRepo %v", repoURL)
+
+	chSrt := &corev1.Secret{}
+
+	if chn.Spec.SecretRef != nil {
+		srtNs := chn.Spec.ConfigMapRef.Namespace
+
+		if srtNs == "" {
+			srtNs = chn.GetNamespace()
+		}
+
+		chnSrtKey := types.NamespacedName{
+			Name:      chn.Spec.SecretRef.Name,
+			Namespace: srtNs,
+		}
+
+		if err := hubClt.Get(context.TODO(), chnSrtKey, chSrt); err != nil {
+			return nil, gerr.Wrapf(err, "failed to get reference secret %v from channel", chnSrtKey.String())
+		}
+	}
+
+	chnCfg := &corev1.ConfigMap{}
+
+	if chn.Spec.ConfigMapRef != nil {
+		cfgNs := chn.Spec.ConfigMapRef.Namespace
+
+		if cfgNs == "" {
+			cfgNs = chn.GetNamespace()
+		}
+
+		chnCfgKey := types.NamespacedName{
+			Name:      chn.Spec.ConfigMapRef.Name,
+			Namespace: cfgNs,
+		}
+
+		klog.V(2).Infof("getting cfg %v from channel %v", chnCfgKey.String(), chn)
+
+		if err := hubClt.Get(context.TODO(), chnCfgKey, chnCfg); err != nil {
+			return nil, gerr.Wrapf(err, "failed to get reference configmap %v from channel", chnCfgKey.String())
+		}
+	}
+
+	httpClient, err := getHelmRepoClient(chnCfg)
+	if err != nil {
+		return nil, gerr.Wrapf(err, "Unable to create client for helm repo %v", repoURL)
+	}
+
+	indexFile, _, err := getHelmRepoIndex(httpClient, sub, chSrt, repoURL)
+	if err != nil {
+		return nil, gerr.Wrapf(err, "unable to retrieve the helm repo index %v", repoURL)
+	}
+
+	helms := make([]*releasev1.HelmRelease, 0)
+
+	for pkgName, chartVer := range indexFile.Entries {
+		releaseCRName, err := utils.PkgToReleaseCRName(sub, pkgName)
+		if err != nil {
+			return nil, gerr.Wrapf(err, "failed to generate releaseCRName of helm chart %v for subscription %v", pkgName, sub)
+		}
+
+		helm, err := utils.CreateOrUpdateHelmChart(pkgName, releaseCRName, chartVer, hubClt, chn, sub)
```

Comment:
There may be override part defined in the subscription, so we probably need to call utils.Override to fresh the helmrelease CR. here is the reference code.

https://github.com/open-cluster-management/multicloud-operators-subscription/blob/2159f30ae0e7e77c01b80814c98c1c1c1cda666a/pkg/utils/helmrepo.go#L259

Reply:
Good call.

I added the override to the `CreateOrUpdateHelmChart`.